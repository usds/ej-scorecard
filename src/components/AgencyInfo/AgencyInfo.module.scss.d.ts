declare namespace AgencyInfoNamespace {
  export interface IAgencyInfoScss {
    agencyInfoCol1: string;
    agencyInfoCol2: string;
    agencyInfoLogo: string;
  }
}

declare const AgencyInfoScssModule: AgencyInfoNamespace.IAgencyInfoScss & {
  /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
  locals: AgencyInfoNamespace.IAgencyInfoScss;
};

export = AgencyInfoScssModule;
