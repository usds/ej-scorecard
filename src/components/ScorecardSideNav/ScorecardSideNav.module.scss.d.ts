declare namespace ScorecardSideNavNamespace {
  export interface IScorecardSideNavScss {
    scorecardSideNavContainer: string;
  }
}

declare const ScorecardSideNavScssModule: ScorecardSideNavNamespace.IScorecardSideNavScss & {
  /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
  locals: ScorecardSideNavNamespace.IScorecardSideNavScss;
};

export = ScorecardSideNavScssModule;
