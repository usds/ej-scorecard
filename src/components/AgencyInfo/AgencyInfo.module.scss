@use "uswds-core" as *;

.agencyInfoCol1{
    display: flex;
    flex-direction: column;
    align-items: center;

};

.agencyInfoCol2{
    display: flex;
    flex-direction: column;
    align-items: flex-start;

    p span {
        display: block;
    }
};

.agencyInfoLogo {
    @include u-margin-left(5);
    @include u-margin-right(5);
    @include u-margin-top(2);
}

