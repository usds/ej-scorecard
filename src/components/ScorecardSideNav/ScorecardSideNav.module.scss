// @use "uswds-core" as *;

// .scorecardSideNavContainer{

// }
