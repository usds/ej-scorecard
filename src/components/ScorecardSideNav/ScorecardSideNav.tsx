import React from 'react';
import { SideNav } from '@trussworks/react-uswds';

// import * as styles from './ScorecardSideNav.module.scss';

import { ScorecardSideNavProps } from '@/types';
import { groupAgencyLinks } from '../AppHeader/AppHeader';
import {
  AGENCY_NAME_RANGE1,
  AGENCY_NAME_RANGE2,
  AGENCY_NAME_RANGE3,
} from '@/data/constants';
import { isPathInGroupRange } from '../util';

const ScorecardSideNav: React.FC<ScorecardSideNavProps> = ({
  allAgencyNames,
  pathname,
}) => {
  if (isPathInGroupRange(pathname, AGENCY_NAME_RANGE1)) {
    return (
      <SideNav items={groupAgencyLinks(allAgencyNames, AGENCY_NAME_RANGE1)} />
    );
  } else if (isPathInGroupRange(pathname, AGENCY_NAME_RANGE2)) {
    return (
      <SideNav items={groupAgencyLinks(allAgencyNames, AGENCY_NAME_RANGE2)} />
    );
  } else if (isPathInGroupRange(pathname, AGENCY_NAME_RANGE2)) {
    return (
      <SideNav items={groupAgencyLinks(allAgencyNames, AGENCY_NAME_RANGE3)} />
    );
  } else {
    return null;
  }
};

export default ScorecardSideNav;
