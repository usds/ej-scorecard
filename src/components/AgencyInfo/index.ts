import AgencyInfo from './AgencyInfo';
export default AgencyInfo;
