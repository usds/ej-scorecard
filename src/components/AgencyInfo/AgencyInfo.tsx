import React from 'react';
import { StaticImage } from 'gatsby-plugin-image';
import { Grid } from '@trussworks/react-uswds';

// import { useAgencyInfo } from '@/hooks/use-agency-info';

import * as styles from './AgencyInfo.module.scss';
import { AgencyInfoProps } from '@/types';
import ScorecardSideNav from '../ScorecardSideNav';

const AgencyInfo: React.FC<AgencyInfoProps> = ({
  info,
  allAgencyNames,
  pathname,
}) => {
  // Todo: may not need this custom hook
  // const agencyInfo = useAgencyInfo();

  return (
    <Grid row gap>
      <Grid desktop={{ col: 8 }} tablet={{ col: 10 }} col={12}>
        <Grid row gap>
          <Grid col={4}>
            <div className={styles.agencyInfoCol1}>
              <StaticImage
                className={styles.agencyInfoLogo}
                src="../../../static/images/agency-a.png"
                alt="Agency logo"
              />
              <h3>{info.Name}</h3>
              <span>{info.Contact_Name}</span>
              <span>{info.Address_Line_1}</span>
              <span>{info.Address_Line_2}</span>
              <span>{info.Phone}</span>
              <span>{info.Site}</span>
            </div>
          </Grid>
          <Grid col={8}>
            <div className={styles.agencyInfoCol2}>
              <h2>{`About ${info.Name}`}</h2>
              <p>{info.About_description}</p>
            </div>
          </Grid>
        </Grid>
      </Grid>
      <Grid desktop={{ col: 1 }} tablet={{ col: 10 }} col={12} />
      <Grid desktop={{ col: 3 }} tablet={{ col: 10 }} col={12}>
        <ScorecardSideNav allAgencyNames={allAgencyNames} pathname={pathname} />
      </Grid>
    </Grid>
  );
};

export default AgencyInfo;
