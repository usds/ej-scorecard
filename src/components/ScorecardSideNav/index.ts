import ScorecardSideNav from './ScorecardSideNav';
export default ScorecardSideNav;
