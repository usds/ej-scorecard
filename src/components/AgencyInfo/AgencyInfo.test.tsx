import createComponentWithIntl from '@/test/testHelpers';
import { mockAllAgencyNames } from '../ScorecardSideNav/ScorecardSideNav.test';
import AgencyInfo from './AgencyInfo';

// import * as Gatsby from 'gatsby';
// import { vi } from 'vitest';

// const useStaticQuery = vi.spyOn(Gatsby, `useStaticQuery`);
// const mockUseStaticQuery = {
//   agencyAEjScorecard2023InfoCsv: {
//     Name: `Agency Name`,
//     Logo: ``,
//     Contact_Name: `Some person`,
//     Address_Line_1: `2020 Washington Ave`,
//     Address_Line_2: `Lincoln, NB, 112233`,
//     Phone: `212-867-5309`,
//     Email: `test@gov.com`,
//     Size: `12`,
//     Value: `$5`,
//     Metrics: `12`,
//     About_description: `Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.`,
//   },
// };

const info = {
  id: `39455749-3310-53aa-b5fa-22590adc1e05`,
  Name: `Agency D`,
  Contact_Name: `Leslie KnopeD`,
  Address_Line_1: `333 Any StreetD`,
  Address_Line_2: `Washington, D.C. 20032`,
  Phone: `(212) 867-5309`,
  Site: `agencya.gov`,
  About_description: `LoremD ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus velit non justo tempus, sit amet bibendum tellus convallis. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nam in gravida mi. Nam nisl ante, ultricies et massa id, euismod sodales arcu. Sed velit felis, tristique a nunc sit amet, rutrum iaculis lectus. Nullam auctor justo at odio sagittis rutrum. Nunc eget malesuada mauris. Donec tincidunt ex in mi ultrices sodales. Aliquam at lobortis eros, eu mollis tellus. Nullam sit amet elementum tellus. Suspendisse viverra ligula neque, id dapibus nisl elementum et. Integer mattis magna ac lorem fermentum, laoreet accumsan tellus tincidunt. Donec bibendum viverra velit vel eleifend.`,
};

describe(`rendering of AgencyInfo Component`, () => {
  // beforeEach(() => {
  //   useStaticQuery.mockImplementation(() => mockUseStaticQuery);
  // });

  // afterEach(() => {
  //   vi.restoreAllMocks();
  // });

  it(`checks if component renders group 1`, () => {
    const component = createComponentWithIntl(
      <AgencyInfo
        info={info}
        allAgencyNames={mockAllAgencyNames}
        pathname={`scorecard/a1-agency-dept-office`}
      />,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`checks if component renders group 2`, () => {
    const component = createComponentWithIntl(
      <AgencyInfo
        info={info}
        allAgencyNames={mockAllAgencyNames}
        pathname={`scorecard/d6-agency-dept-office`}
      />,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`checks if component renders group 3`, () => {
    const component = createComponentWithIntl(
      <AgencyInfo
        info={info}
        allAgencyNames={mockAllAgencyNames}
        pathname={`scorecard/t1-agency-dept-office`}
      />,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
