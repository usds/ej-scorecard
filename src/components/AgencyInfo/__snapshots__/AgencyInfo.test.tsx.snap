// Vitest Snapshot v1

exports[`rendering of AgencyInfo Component > checks if component renders 1`] = `
<div
  className="grid-row grid-gap"
  data-testid="grid"
>
  <div
    className="grid-col-12 tablet:grid-col-10 desktop:grid-col-8"
    data-testid="grid"
  >
    <div
      className="grid-row grid-gap"
      data-testid="grid"
    >
      <div
        className="grid-col-4"
        data-testid="grid"
      >
        <div>
          <h3>
            Agency D
          </h3>
          <span>
            Leslie KnopeD
          </span>
          <span>
            333 Any StreetD
          </span>
          <span>
            Washington, D.C. 20032
          </span>
          <span>
            (212) 867-5309
          </span>
          <span>
            agencya.gov
          </span>
        </div>
      </div>
      <div
        className="grid-col-8"
        data-testid="grid"
      >
        <div>
          <h2>
            About Agency D
          </h2>
          <p>
            LoremD ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus velit non justo tempus, sit amet bibendum tellus convallis. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nam in gravida mi. Nam nisl ante, ultricies et massa id, euismod sodales arcu. Sed velit felis, tristique a nunc sit amet, rutrum iaculis lectus. Nullam auctor justo at odio sagittis rutrum. Nunc eget malesuada mauris. Donec tincidunt ex in mi ultrices sodales. Aliquam at lobortis eros, eu mollis tellus. Nullam sit amet elementum tellus. Suspendisse viverra ligula neque, id dapibus nisl elementum et. Integer mattis magna ac lorem fermentum, laoreet accumsan tellus tincidunt. Donec bibendum viverra velit vel eleifend.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
